stetho/src/main/java/com/facebook/stetho/inspector/network/AsyncPrettyPrinterSchemaManager.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.network;
+
+import javax.annotation.Nullable;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class AsyncPrettyPrinterSchemaManager {
+  private static AsyncPrettyPrinterSchemaManager sSchemaManager;
+  private Map<URL, String> mSchemaCache;
```

Comment:
I just realized you are threading the `URL` all the way through. `URL#equals()` will actually perform dns lookup to verify the hosts resolve to the same address. It would probably better to take the String representation of the url and use that. Its okay to keep passing in URL but I would make this a `Map<String, String>` and use `URL.toString()` when calling `put` or `get`